app/shared-components/PromoModal/PromoModal.jsx

```diff
@@ -0,0 +1,201 @@
+/**
+ * Base component for all of the Promo Modals
+ *
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2019 Ghostery, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import ClassNames from 'classnames';
+import Modal from '../Modal';
+import InsightsPromoModalContent from '../ModalContent/InsightsPromoModalContent';
+import PlusPromoModalContent from '../ModalContent/PlusPromoModalContent';
+import PremiumPromoModalContent from '../ModalContent/PremiumPromoModalContent';
+import history from '../../panel/utils/history';
+import { sendMessage } from '../../panel/utils/msg';
+import globals from '../../../src/classes/Globals';
+import ModalExitButton from '../../panel/components/BuildingBlocks/ModalExitButton';
+
+const DOMAIN = globals.DEBUG ? 'ghosterystage' : 'ghostery';
+const INSIGHTS = 'insights';
+const PLUS = 'plus';
+const PREMIUM = 'premium';
+
+/**
+ * A base functional component for Promo Modals
+ * @return {JSX}
+ * @memberof HubComponents
+ */
+
+class PromoModal extends React.Component {
+	/**
+	 * @param modal			'insights' or 'premium'
+	 * @private
+	 * Handle clicks on the link to turn off promos in the promo modals
+	 */
+	_handlePromoGoAwayClick = (modal) => {
+		this.props.actions.togglePromoModal();
+
+		sendMessage('promoModals.turnOffPromos', {});
+
+		switch (modal) {
+			case INSIGHTS:
+				sendMessage('ping', 'promo_modals_decline_insights_upgrade');
+				break;
+			case PLUS:
+				sendMessage('ping', 'promo_modals_decline_plus_upgrade');
+				break;
+			case PREMIUM:
+				sendMessage('ping', 'promo_modals_decline_insights_upgrade');
```

Comment:
Should this be `.....decline_premium_upgrade`?

Zooming out, perhaps instead of a switch we can simply send `promo_modals_decline_${modal}_upgrade`?

Reply:
It looks like we currently don't have a metrics ping case for this so I reached out to Teresa and she said to hold off on adding extra pings for now since wants to sync with JJ first. I refactored the plus and insights pings to use the template literal notation for now.